security-jwt/src/main/java/io/micronaut/security/token/jwt/signature/rsa/RSASignatureGenerator.java

```diff
@@ -64,7 +69,8 @@ public SignedJWT sign(JWTClaimsSet claims) throws JOSEException {
      */
     protected SignedJWT signWithPrivateKey(JWTClaimsSet claims, @NotNull RSAPrivateKey privateKey) throws JOSEException {
         final JWSSigner signer = new RSASSASigner(privateKey);
-        final SignedJWT signedJWT = new SignedJWT(new JWSHeader(algorithm), claims);
+        JWSHeader jwsHeader = keyId == null ? new JWSHeader(algorithm) : new JWSHeader.Builder(algorithm).keyID(keyId).build();
```

Comment:
Could you just use the builder here? Setting the `keyID` to null should have no impact